bundles/org.openhab.core/src/main/java/org/openhab/core/items/MetadataRegistry.java

```diff
@@ -36,10 +38,17 @@ public interface MetadataRegistry extends Registry<Metadata, MetadataKey> {
      */
     boolean isInternalNamespace(String namespace);
 
+    /**
+     * Provides all namespaces of a particular item
+     *
+     * @param itemname the name of the item for which the namespaces should be searched.
+     */
+    public Set<String> getAllNamespaces(String itemname);
```

Comment:
```suggestion
    public Collection<String> getAllNamespaces(String itemname);
```

I don't think there is a need to be more specific.